### PR TITLE
fixes stop condition on sync emulator

### PR DESCRIPTION
**Pull request by anton-christensen**

### CODE REVIEW

File: emulators/SyncEmulator.py [hunk 1]
```diff
@@ -106,8 +108,10 @@ def print_statistics(self):
 
     def terminated(self, index:int):
         self._progress.acquire()
+        self._done[index] = True
         self._terminated += 1
-        if self._terminated >= len(self._devices):
+        if all([self._done[x] or not self._threads[x].is_alive()
+                for x in self.ids()]):
             if self._round_lock.locked():
```
Comment (petergjoel) [inline emulators/SyncEmulator.py:115]: @anton-christensen is there some indentation that has gone wrong here?

File: emulators/SyncEmulator.py [hunk 2]
```diff
@@ -106,8 +108,10 @@ def print_statistics(self):
 
     def terminated(self, index:int):
         self._progress.acquire()
+        self._done[index] = True
         self._terminated += 1
```
Comment (petergjoel) [inline emulators/SyncEmulator.py:112]: I think this `_terminated` variable should die now, the `_done` array implements the same functionality (but in a more robust way)

**Review by petergjoel — CHANGES_REQUESTED** [review]

File: emulators/SyncEmulator.py (hunk 1 above)
Comment (anton-christensen) [reply emulators/SyncEmulator.py:112]: Small stylistic choice, but its a short line so i've moved it onto the same line.

File: emulators/SyncEmulator.py (hunk 2 above)
Comment (anton-christensen) [reply emulators/SyncEmulator.py]: Yup, it's removed in the new commit

**Review by petergjoel — APPROVED** [review]

**Pull request merged**